### ISSUE TITLE
Getting error in profiling, _ProfiledThreadingLock object has no attribute _self_acquired_at

### ISSUE THREAD

**Issue by akira** [DataDog/dd-trace-py#11343]
Upgraded to the most recent release of ddtrace, 2.16.0.  

We are now seeing this error on profiling:
```
Error recording lock release event: '_ProfiledThreadingLock' object has no attribute '_self_acquired_at'
```

We are seeing these errors, but it also appears to impact the applications as well (queue workers no longer dequeuing tasks).  We are using asyncio with uvloop (0.19.0) and Python 3.11.10.

**Comment by akira** [comment]
Any update on this?  This is preventing usage of the profiler for us in the latest version

**Comment by dannosaur** [comment]
Yep, we're seeing this too. For some reason, only on our celerybeat service.

Python 3.11.9, ddtrace 2.11.8.

**Comment by bvs-langchain** [comment]
I'm seeing a different but seemingly related error with `2.16.0`:

```
ddtrace.profiling.collector._lock: Failed to get lock variable name, we only support local/global variables and their attributes.
```

**Comment by audiolion** [comment]
We are also seeing this error:

```
Failed to get lock variable name, we only support local/global variables and their attributes.
```

On `2.18.1` with python `3.10`

**Comment by taegyunkim** [comment]
The fix PR was merged but has not yet been released. Will update once more when it's released.

**Comment by taegyunkim** [comment]
@audiolion 
https://github.com/DataDog/dd-trace-py/blob/taegyunkim/prof-11111-v2-on/ddtrace/profiling/collector/_lock.py#L296

It's just a debug log. Are your services spammed with this message?

**Comment by danyi1212** [comment]
Hey @taegyunkim, starting on version 2.20.0, when we enable the Profiler, we start getting this exact exception:
```python 
builtins.AttributeError: '_ProfiledThreadingLock' object has no attribute '_self_acquired_at'
```
When we reverted to 2.19.1 it was solved.

Wasn't it supposably solved in this exact version? 🤔
https://github.com/DataDog/dd-trace-py/releases/tag/v2.20.0#:~:text=Resolves%20a%20data%20race%20issue%20accessing
